lib/daru/clean/clean.rb

```diff
@@ -0,0 +1,76 @@
+module Daru
+  # Module encapsulating advanced data cleaning methods on DataFrame.
+  module Clean
+    # replace elements with those from given set of elements
+    # based on similarity
+    # @param vector [Daru::Vector] vector to be cleansed
+    # @param dict [Array] the array serving as key set
+    # @example
+    #   df = Daru::DataFrame.new({ a: ['mal','femal','M','F','Fem']})
+    #   df.fuzzy_match :a, ['MALE','FEMALE']
+    #   # => #<Daru::DataFrame(5x1)>
+    #   #           a
+    #   #   0    MALE
+    #   #   1  FEMALE
+    #   #   2    MALE
+    #   #   3  FEMALE
+    #   #   4  FEMALE
+    def fuzzy_match vector, dict
+      # TODO
+    end
+
+    # fill missing places using linear regression
+    # @param vector [Daru::Vector] vector to be cleansed
+    # @example
+    #   df = Daru::DataFrame.new({ a: [1,3,nil,7,9,nil]})
+    #   df.impute :a
+    #   # => #<Daru::DataFrame(6x1)>
+    #   #       a
+    #   #   0   1
+    #   #   1   3
+    #   #   2   5
+    #   #   3   7
+    #   #   4   9
+    #   #   5   11
+    def impute vector
+      # TODO
+    end
+
+    # Split multi valued cells in more than one column into rows
```

Comment:
The example looks useful, yet underthought:
* how do we specify which columns to split?
* how do we specify what is split substring? (I believe `,` is not the only possibility)
* split_**cell** (singular) looks like not the best name possible.

Reply:
I feel letting the user specify the details would make them use the method less likely (which is the case in the corresponding method in OpenRefine). Although modularity does no harm, I was thinking of applying split to all the noticeable columns with such error. I will update the doc with an option for specifying the delimiter and a better method name.